Filebeat integration test with new rats

ph opened the pull request:
This PR uses the new framework developed by @suyograo to add integration test.
Few things in done in this pull request:

- Add support for multiple named config in the yaml files.
- Add support for ERB in the YAML files
- Make the ServiceLocator more flexible and use the file name instead of the predefined hash.
- Add a gitignore
- Add keys for the tests and a generator.
- Add test for theses scenario:
  - No TLS
  - TLS Server Auth
  - TLS Mutual Auth

suyograo commented on logstash-core-event-java/lib/logstash-core-event-java_jars.rb:
```diff
@@ -3,12 +3,12 @@
 rescue LoadError
   require 'com/fasterxml/jackson/core/jackson-databind/2.7.3/jackson-databind-2.7.3.jar'
-  require 'com/fasterxml/jackson/core/jackson-annotations/2.7.3/jackson-annotations-2.7.3.jar'
+  require 'com/fasterxml/jackson/core/jackson-annotations/2.7.0/jackson-annotations-2.7.0.jar'
```
Why this change?

ph replied:
I think it got fetch when I ran the rake test:install-core I will revert it.

suyograo commented on logstash-core-event-java/lib/logstash-core-event-java_jars.rb:
```diff
@@ -3,12 +3,12 @@
 if defined? Jars
   require_jar( 'com.fasterxml.jackson.core', 'jackson-databind', '2.7.3' )
-  require_jar( 'com.fasterxml.jackson.core', 'jackson-annotations', '2.7.3' )
+  require_jar( 'com.fasterxml.jackson.core', 'jackson-annotations', '2.7.0' )
```
Can we go back to 2.7.3?

suyograo commented:
We need to remove `qa/integration/fixtures/how.input` from this PR. You can add a script to download it or can use a subset of it. See https://github.com/elastic/logstash/blob/feature/ruby_integration_tests/qa/integration/fixtures/kafka_input.input. 

Maybe we can rename this file to `how_sample.input` and reuse them in multiple tests?

suyograo commented on qa/integration/fixtures/beats_input_spec.yml:
```diff
@@ -0,0 +1,33 @@
+        port => 5044
+      }
+    }
+  tls_server_auth: |-
```
nice. Keyed config!

suyograo commented on qa/integration/fixtures/beats_input_spec.yml:
```diff
@@ -0,0 +1,33 @@
+      beats {
+        ssl => true
+        port => 5044
+        ssl_certificate => '<%=options[:ssl_certificate]%>'
```
Awesome `erb` addition

suyograo commented on qa/integration/services/service_locator.rb:
```diff
@@ -1,19 +1,30 @@
+    @services.fetch(name)
+  end
+
+  def available_services
```
Nice!

ph replied:
Hash#fetch is awesome we should use it more.. with it its over looking all over the place when you have a nil value.

suyograo commented:
@ph just a couple of concerns and this is good to go. Thanks for adding the enhancements!

jsvd commented:
Can we somehow avoid including that how.input file in the repo? an uncompressed 64 mb file shouldn't be commited to git

jsvd commented on qa/integration/services/filebeat_setup.sh:
```diff
@@ -0,0 +1,26 @@
+  echo "Filebeat version is $FILEBEAT_VERSION"
+  version=$FILEBEAT_VERSION
+else
+   version=5.0.0-alpha5
```
if possible I suggest picking this version from the versions.yml file, so we don't have to maintain this :)

ph replied:
true, they should match now.

ph commented:
@suyograo I update this PR with the following:
- removed `how` data,
- renamed `kafka_input.input` to `how_sample.input`
- reverted the jackson changes.

I haven synchronized the versions.yml with the bash script, I think we can do it in the feature branch. WDYT?

suyograo commented:
LGTM
